### PR TITLE
fix(TDI-46171): support custom connection/close component icon for tck connector in studio

### PR DESCRIPTION
**Pull request by wwang-talend**
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant 

### Why this PR is needed?
https://jira.talendforge.org/browse/TCOMP-2552

### What does this PR adds (design/code thoughts)?
https://jira.talendforge.org/browse/TCOMP-2552

### CODE REVIEW

**Comment by wwang-talend** [comment]
sonar advice fix, and not care the duplicated string issues which exists before, need more careful refactor.

File: component-server-parent/component-server-api/src/main/java/org/talend/sdk/component/server/api/ComponentResource.java [hunk 1]
```diff
@@ -135,6 +135,23 @@ Response icon(@PathParam("id") @Parameter(name = "id",
             description = "Component icon identifier.",
             in = PATH) String id);
 
+    @GET
+    @Path("searchIcon/{familyId}/{iconKey}")
```
Comment (undx) [inline component-server-parent/component-server-api/src/main/java/org/talend/sdk/component/server/api/ComponentResource.java:139]: I tend to prefer not changing api prefix, rather use `icon`. smtg like `icon/custom/...`.
```suggestion
    @Path("icon/custom/{familyId}/{iconKey}")
```

Comment (wwang-talend) [reply component-server-parent/component-server-api/src/main/java/org/talend/sdk/component/server/api/ComponentResource.java]: @undx fixed

**Review by undx — DISMISSED** [review]
lgtm

**Review by undx — APPROVED** [review]

**Pull request merged**